caffe2/operators/slice_op.h

```diff
@@ -25,12 +25,85 @@ namespace caffe2 {
 
 namespace {
 
+template<class SIndex, class Context>
+void slice_(
```

Comment:
this is cool. For GPU I think the right model would be instead of recursion doing the iteration in the 'variable number system'. I.e. if you have 3 dimensions [a,b,c] after collapsing, run the loop and do `x = i % a; y = i/a%b; z = i/a/b`. On GPU that's run in parallel and execute super-fast. The current implementation does pretty much the same but only for one dimension.